src/Controller/Component/CategoriesComponent.php

```diff
@@ -37,12 +37,24 @@ public function initialize(array $config)
      * Load categories by object type.
      *
      * @param string|null $type Object type name.
+     * @param string|null $parentId ID of parent category.
      * @return \Cake\ORM\Query
      */
-    public function load(?string $type = null): Query
+    public function load(?string $type = null, ?string $parentId): Query
     {
         $query = $this->Categories->find()
-            ->where([$this->Categories->aliasField('enabled') => true])
+            ->where([$this->Categories->aliasField('enabled') => true]);
+
+        if ($parentId !== null) {
+            $query = $query
+                ->where([$this->Categories->aliasField('parent_id') => $parentId]);
+        }
+
+        if ($type !== null) {
+            $query = $query->find('type', [$type]);
+        }
```

Comment:
Query objects are lazily executed, so it was perfectly fine to add `where()` conditions and append finders after setting `order()` and results formatter. However, if deemed clearer, it's perfectly fine to group conditions first. 👍🏼

Reply:
It seems clearer to me to first filter by given filters, if any.